pybamm/solvers/base_solver.py

```diff
@@ -1160,11 +1160,34 @@ def step(
             raise pybamm.SolverError(
                 f"Step time must be at least {pybamm.TimerTime(step_start_offset)}"
             )
-
         t_start = old_solution.t[-1]
-        t_end = t_start + dt
-        # Calculate t_eval
-        t_eval = np.linspace(t_start, t_end, npts)
+        t_end = t_start
```

Comment:
should `t_end` be `t_start+dt`?

Reply:
in fact, you don't need to define this here?

Reply:
That's what I did before but removing the line throws this error :
```
  t_end = t_end + time_diff
UnboundLocalError: local variable 't_end' referenced before assignment
```
Maybe `t_end` needs to be initialized before.

Reply:
@rtimms what should be done here?

Reply:
Maybe you can try doing `t_end = t_start + dt + time_diff` instead of `t_end = t_end + time_diff` in [L1182](https://github.com/pybamm-team/PyBaMM/pull/3627/files?diff=unified&w=0#diff-630d9645dc8c18de32e60b33d118b8dc8c243a8641df8a65fb4ad971b96f6064R1182)

Reply:
see above comment